### PR TITLE
implement mint conversions

### LINKED ISSUE #314: Mint Integration!
**Is your feature request related to a problem? Please describe.**
`egui` has been an awesome solution to getting ui up and running for for my game. For all of my coordinate systems, I'm currently using `glam` and would love an easy way to translate some of my vectors easily into egui's format without having to unpack them every time.

**Describe the solution you'd like**
Fortunately, that's why Mint exists: https://github.com/kvark/mint

Mint is a data / representation only linear algebra library meant to be used as a unified intermediate step when converting between linear algebra representations. All each library has to do is implement `from<mint::types>` for their own representations, and users can easily convert between them.

Here's a short list of libs that implement mint conversions:
- `cgmath`
- `euclid`
- `nalgebra`
- `glam`
- `kurbo`
- `vek`

The library itself is fairly lightweight, (7.5kb), has a single (optional) dependency on `serde` and could be enabled / disabled with a feature easily.

**Describe alternatives you've considered**
To my knowledge `mint` is the best supported interoperability lib for math-related types. If `egui` had consumes another library's types, I would suggest to just re-export them. But `egui` wrote them from scratch, so the next best solution would be to implement conversions for `mint`

This would be a pretty lightweight feature, and would be happy to contribute it, I just wanted to raise the idea first, and gauge how open y'all are to the idea. 

In general, I would just like to thank y'all for the work on this lib. It's been a pleasure to use, and I hope to improve it even more.

### PR DESCRIPTION
**Pull request by LU15W1R7H**
Implement conversions for [mint](https://docs.rs/mint) (math interoperability standard types).

- `impl {From, Into}<mint::Point2> for Pos2`
- `impl {From, Into}<mint::Vector2> for Vec2`

Closes #314

### CODE REVIEW

File: emath/src/pos2.rs [hunk 1]
```diff
@@ -84,6 +84,26 @@ impl From<&Pos2> for (f32, f32) {
     }
 }
 
+// ----------------------------------------------------------------------------
+// Compatibility and convenience conversions to and from (f32, f32):
```
Comment (emilk) [inline emath/src/pos2.rs:88]: rm misleading comment 

```suggestion
```

File: emath/src/vec2.rs [hunk 2]
```diff
@@ -81,6 +81,26 @@ impl From<&Vec2> for (f32, f32) {
     }
 }
 
+// ----------------------------------------------------------------------------
+// Compatibility and convenience conversions to and from (f32, f32):
```
Comment (emilk) [inline emath/src/vec2.rs:85]: rm misleading comment 

```suggestion
```

**Review by emilk — CHANGES_REQUESTED** [review]
Cool!

So an `egui` user would need to add `emath = { version = 0.11, features = ["mint"] }` to turn this on, correct? Would it maybe make sense to add `mint` as a feature in `egui` too, so that users could just change `egui = "0.11"` into `egui = { version = "0.11", features = ["mint"] }` without having to know about `emath`?

File: emath/src/pos2.rs (hunk 1 above)
Comment (LU15W1R7H) [reply emath/src/pos2.rs:88]: Sorry, forgot to adjust the comment 😅

**Comment by LU15W1R7H** [comment]
I fixed the clippy issue.

**Review by emilk — APPROVED** [review]
Thanks!

**Pull request merged**